129 content managers can add buttons to blocks

MariaBee3 opened the pull request:
This pull request adds the possibility to the CMS for content managers to add links as buttons to several blocks: 
- Title block
- Text-Media block
- Hero block
- Cards block

Links can go to an internal page or an external page. 
Content managers can add one or multiple buttons and choose if they want to outline them left or centered.

antw reviewed (commented):
This looks good to me.

My only concern is that the `Button` and `Buttons` types are repeated several times. It's better to define a type in one place and reuse it. Otherwise, if the data changes in the future (Wagtail returns additional, or fewer, attributes) you'll have to hunt through many different files and change them all.

There are two possibilities:

1. Define a `types.ts` file somewhere with common types and import from this file whenever you need the types:

    ```ts
    import type { Button, Buttons } from "../components/types"
    ```

2. Alternatively, see if you can use `React.ComponentProps` to get the button type from the Button component. For example:

    ```ts
    // components/HeroBlock/HeroBlock.tsx

    import ButtonBlock from "@/components/Button/ButtonBlock"

    type Props {
      data: {
        // ...
        value: {
          // ...
          // This gets the type of "buttons" from the ButtonBlock props
          buttonBlock: React.ComponentProps<ButtonBlock>["buttons"]
        }
      }
    };
    ```

    This has the advantage that you whenever you change the type for the ButtonBlock props, that change is immediately reflected in all the other components.

MariaBee3 commented:
@antw Thank you so much for this suggestion. This was also bothering me, I was hoping you would have a good solution for this. I implemented the second solution and also added it somewhere for the mediacontent component. Thanks again.
